File: API/Helpers/UserParams.cs
﻿ namespace API.Helpers
{
    public class UserParams : PaginationParams
    {
        public string CurrentUsername { get; set; }
        public string InterestedIn { get; set; }
        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 150;
        public string OrderBy { get; set; } = "lastActive";
    }
}

File: client/src/app/_models/userParams.ts
import { User } from "./user";

export class UserParams{
    interestedIn:string;
    minAge = 18;
    maxAge=99;
    pageNumber=1;
    pageSize=5;
    orderBy ='lastActive';

    constructor(user: User){
        this.interestedIn = user.gender === 'female'?'male':'female';
    }
}